server.go

```diff
@@ -143,10 +160,14 @@ func (s *Server) Open() error {
 		return fmt.Errorf("opening NodeSet: %v", err)
 	}
 
+	// Create default HTTP client
+	s.createDefaultClient()
+
 	// Create executor for executing queries.
-	e := NewExecutor()
+	e := NewExecutor(&ClientOptions{TLS: s.TLS})
```

Comment:
I think that ClientOptions shouldn't be an argument to NewExecutor. The pattern we have been following is to call NewExecutor and set options on the resulting object (below). Perhaps a method `Executor.SetClientOptions(*ClientOptions)` would set everything up (including Scheme and Host)?

Reply:
`ClientOptions` contains configuration for only the client which is used to access other nodes (like syncing blocks, etc.). `Host` and `Scheme` belong to the node which the executor is running, so I don't think it makes sense to put them in client options. Maybe you meant to have something like `ExecutorOptions` which would include `ClientOptions` ?

`NewExecutor` sets up the HTTP client so IMO it's better to pass client options (or executor options?) to it. Is there an advantage of passing nothing to `NewExecutor` and having separate `Set` functions ? If the user wants to initialize an executor they can always use `Executor{}`.